StripePaymentSheet/StripePaymentSheet/Source/PaymentSheet/IntegrationConfigurable.swift

```diff
@@ -0,0 +1,63 @@
+//
+//  IntegrationConfigurable.swift
+//  StripePaymentSheet
+//
+//  Created by Nick Porter on 10/1/24.
+//
+
+import Foundation
+import UIKit
+@_spi(STP) import StripePayments
+
+/// Represents shared configuration properties shared between integration surfaces in mobile payment element.
+/// - Note: See the concrete implementations of `IntegrationConfigurable` for detailed doc comments.
+/// - Note: Not currently used by CustomerSheet.
+protocol IntegrationConfigurable: PaymentMethodRequirementProvider {
```

Comment:
Open to naming suggestions

Reply:
Let's call it PaymentElementConfiguration? I think that's the most accurate name, since this is Configuration for all Mobile Payment Element integration variants.  We just need to start thinking of PaymentSheet, FlowController, EmbeddedPaymentElement as Payment Element variants.

Reply:
This means we should probably rename things like `PaymentSheetLoader` to be `PaymentElementLoader`.

Reply:
Agree on renaming other PaymentSheet prefixed classes, maybe something we should tackle when we rename the module.